src/pint/toa.py

```diff
@@ -1639,6 +1639,37 @@ def get_flag_value(self, flag, fill_value=None, as_type=None):
             result.append(val)
         return result, valid_index
 
+    def set_flag_values(self, flagname, flagvalues):
```

Comment:
Could this just be implemented by letting `__setitiem__` accept arrays?

Reply:
I don't think so.  It would actually be nice to just allow:
```
t['flags']['key']
```
but I don't think that works with the current data model.  But maybe I'm wrong.  There is also the need to turn the flags into strings, but avoid(?) string arrays.

Reply:
Don't we allow `t['-key']` as if a flag were a column? I agree string arrays are awkward.

Reply:
I guess I didn't realize that `t['key']` would work (I had tried `t['flags']['key']`).  But given the issues around string arrays and other format conversions, I still think this function is useful.

Reply:
Would it be equally effective to make `t["key"]` apply `str` to each element of its argument? Possibly except if they're None... I'm trying to avoid something like the mess on TimingModels where there are heaps of different ways to set things and no one is sure which ones work.

Reply:
Yes, that might work.  As I said I didn't know about this syntax: I use `get_flag_value()` to get results, and that gives slightly different results to just `t["key"]` (you get to specify what the missing values are, and what the resulting data-type is).  But if there were a way to set values more easily that would be useful.

Reply:
Yes, `.get_flag_values()` is analogous to a dictionary's `.get()` function, that allows a default value instead of the exception produced by indexing. But I'm not sure we need two different ways of setting flag values.